plugins/chat/assets/javascripts/discourse/components/chat-thread-list.gjs

```diff
@@ -1,15 +1,111 @@
 import Component from "@glimmer/component";
 import { bind } from "discourse-common/utils/decorators";
 import { cached } from "@glimmer/tracking";
-import { action } from "@ember/object";
 import { inject as service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import ChatThreadListHeader from "discourse/plugins/chat/discourse/components/chat/thread-list/header";
+import ChatThreadListItem from "discourse/plugins/chat/discourse/components/chat/thread-list/item";
+import eq from "truth-helpers/helpers/eq";
+import I18n from "I18n";
+import ChatTrackMessage from "discourse/plugins/chat/discourse/modifiers/chat/track-message";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
+import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
 
 export default class ChatThreadList extends Component {
+  <template>
+    {{! template-lint-disable modifier-name-case }}
+    {{#if this.shouldRender}}
+      <div class="chat-thread-list" {{this.subscribe @channel}}>
+        {{#if @includeHeader}}
+          <ChatThreadListHeader @channel={{@channel}} />
+        {{/if}}
+
+        <div class="chat-thread-list__items" {{this.fill}}>
+          {{#each this.sortedThreads key="id" as |thread|}}
+            <ChatThreadListItem
+              @thread={{thread}}
+              {{(if
+                (eq thread this.lastThread)
+                (modifier ChatTrackMessage this.load)
+              )}}
+            />
+          {{else}}
+            {{#if this.threadsCollection.fetchedOnce}}
+              <div class="chat-thread-list__no-threads">
+                {{this.noThreadsLabel}}
+              </div>
+            {{/if}}
+          {{/each}}
+
+          <ConditionalLoadingSpinner
+            @condition={{this.threadsCollection.loading}}
+          />
+
+          <div {{this.loadMore}}>
+            <br />
+          </div>
+        </div>
+      </div>
+    {{/if}}
+  </template>
+
   @service chat;
   @service chatApi;
   @service messageBus;
   @service chatTrackingStateManager;
 
+  noThreadsLabel = I18n.t("chat.threads.none");
+
+  subscribe = modifier((element, [channel]) => {
+    this.messageBus.subscribe(
+      `/chat/${channel.id}`,
+      this.onMessageBus,
+      channel.channelMessageBusLastId
```

Comment:
This is not visible due to the gjs move but there's actually a very important fix here. It's not related to the PR, but the pr was flakey because of this bug: we were using `channel.messageBusLastId` which is undefined and was causing incorrect tracking.